pkg/querier/blocks_store_queryable.go

```diff
@@ -546,7 +547,7 @@ func (q *blocksStoreQuerier) fetchSeriesFromStores(
 	return seriesSets, queriedBlocks, warnings, int(numChunks.Load()), nil
 }
 
-func createSeriesRequest(minT, maxT int64, matchers []storepb.LabelMatcher, blockIDs []ulid.ULID) (*storepb.SeriesRequest, error) {
+func createSeriesRequest(minT, maxT int64, matchers []storepb.LabelMatcher, sp *storage.SelectHints, blockIDs []ulid.ULID) (*storepb.SeriesRequest, error) {
```

Comment:
I would pass only a `skipChunks boolean` instead of the whole `sp`. The reason is that we should never read mint/maxt from `sp *storage.SelectHints` and always use the input `minT/maxT`. If we don't pass `sp *storage.SelectHints`, then we would avoid any mistake.